libbeat/docs/shared-kibana-config.asciidoc

```diff
@@ -0,0 +1,213 @@
+
+You configure the endpoint in the `setup.kibana` section of the
++{beatname_lc}.yml+ config file.
+
+Here is an example configuration:
+
+[source,yaml]
+----
+setup.kibana.host: "localhost:5601"
+----
+
+[float]
+=== Configuration options
+
+You can specify the following options in the `setup.kibana` section of the
++{beatname_lc}.yml+ config file:
+
+//REVIEWERS: I've decided to use full paths when referring to options rather than using indentation in examples. WDYT? Makes the examples a tad harder to read but will help immensely with users who don't know where to put stuff in the yaml hierarchy.
+
+[float]
+==== `setup.kibana.host`
+
+The Kibana host where the dashboards will be loaded. This setting is required.
+The scheme and port default to `http` and `5601` if you don't set them
+explicitly. If you use a path after the port number, you must specify
+the scheme and port. For example: `http://localhost:5601/path`.
+
+IPv6 addresses must be defined using the following format:
+`https://[2001:db8::1]:5601`.
+
+[float]
+==== `setup.kibana.protocol`
+
+The name of the protocol Kibana is reachable on. The options are: `http` or
+`https`. The default is `http`. However, if you specify a URL for host, the
+value of `protocol` is overridden by whatever scheme you specify in the URL.
+
+//REVIEWERS: Please check this example. I have not tested it.
+
+Example config:
+
+[source,yaml]
+----
+setup.kibana.host: "192.0.2.255:5601"
+setup.kibana.protocol: "https"
+setup.kibana.path: /kibana
+----
+
+
+[float]
+==== `setup.kibana.username`
+
+The basic authentication username for connecting to Kibana.
+
+[float]
+==== `setup.kibana.password`
+
+The basic authentication password for connecting to Elasticsearch.
+
+[float]
+==== `setup.kibana.path`
+
+An HTTP path prefix that is prepended to the HTTP API calls. This is useful for
+the cases where Kibana listens behind an HTTP reverse proxy that exports the API
+under a custom prefix.
+
+[float]
+==== `setup.kibana.ssl.enabled`
+
+Enables {beatname_uc} to use SSL settings when connecting to Kibana via HTTPS.
+
+//REVIEWERS: The reference config file said that the ssl.enabled: true is the default, but that doesn't seem quite right. Do you mean that this setting defaults to true when https is configured?
+
+Example configuration:
+
+//REVIEWERS: Please confirm this example. I did not test it.
+
+[source,yaml]
+----
+setup.kibana.host: "192.0.2.255:5601"
+setup.kibana.protocol: "https"
+setup.kibana.ssl.enabled: true
+setup.kibana.ssl.certificate_authorities: ["/etc/pki/root/ca.pem"]
+setup.kibana.ssl.certificate: "/etc/pki/client/cert.pem"
+setup.kibana.ssl.key: "/etc/pki/client/cert.key
+----
+
+[float]
+==== `setup.kibana.ssl.verification_mode`
+
+//REVEIWERS: Most of this content is copied from the section about configuring outputs. We should probably centralize our descriptions of these options at some point, but copy/paste is faster, so I'm doing that for now
```

Comment:
Yes, ideally would be not to copy it from here: https://www.elastic.co/guide/en/beats/filebeat/6.x/configuration-output-ssl.html. In case something changes, we can change it in a single place.